src/Schemas/helpers.ts

```diff
@@ -0,0 +1,89 @@
+import { z, ZodSchema } from 'zod/v4';
+import { $ZodError } from 'zod/v4/core';
+
+// TODO: maybe we could have a more descriptive name for this function
+// TODO: we should also make this more generic so that it works on objects too
+// i.e. it might be nice to check an array of users and make sure that the usernames
+// aren't duplicated
+export const unique = (key: string) => {
+  return (data: string[], ctx: z.RefinementCtx) => {
+    if (!data || data.length == 0) {
+      return;
+    }
+
+    const seen = new Set();
+    data.forEach((item, index) => {
+      if (seen.has(item)) {
+        ctx.addIssue({
+          code: 'custom',
+          message: `The ${key} items must be unique`,
+          path: [index],
+        });
+      } else {
+        seen.add(item);
+      }
+    });
+  };
+};
+
+export const unwrapField = (error: $ZodError) => {
+  return error.issues.map((issue) => issue.message);
+};
+
+export const validateField = <T>(value: T, Schema: ZodSchema<T>) => {
+  const result = Schema.safeParse(value);
+
+  if (result.success) {
+    return [];
+  }
+
+  return unwrapField(result.error);
+};
+
+export const unwrapForm = (error: $ZodError) => {
```

Comment:
Oh niooo, I had to write a very similar function for my thesis. This is quite complex though - I know there is no different way than do this while looking complex.
Why do we need this? I cannot tell from the context, and it makes the readability a bit worse. But I am pretty sure that we can make this both readable and safe.